app/controllers/sessions_controller.rb

```diff
@@ -1,135 +1,112 @@
+              <br>Please enter a valid redeemify code.'.html_safe
           render :new ; return
-        end  
-      end
-        #setting up vendor codes for current user
-        @current_code = current_user.code
-        vendors = Vendor.all
-        vendors.each do |vendor|
-          code = vendor.serve_code current_user
-          flash.now[:alert] = 'Some offers\' code are not available at this time, please come back later' unless code
-          
-          @list_codes[vendor.name] ||= code ? code.code : "Not Available"
-          @instruction[vendor.name] ||= vendor.instruction
-          @help[vendor.name] ||= vendor.helpLink
-          @expiration[vendor.name] ||= vendor.expiration
-          @website[vendor.name] ||= vendor.website
-          @cashValue[vendor.name] ||= vendor.cashValue
-          @description[vendor.name] ||= vendor.description
-          @total += vendor.cashValue.gsub(/[^0-9\.]/,'').to_f
-          @total = @total.round(2)
         end
+      end
+      set_vendor_code(current_user)
     end
-    if session[:vendor_id] != nil
+    if session[:vendor_id].present?
       @vendor_user = session[:vendor_id]
     end
   end
-
-
-
+  
   def delete_page
-    
   end
 
   def delete_account
-    if current_user != nil
+    if current_user.present?
       current_user.anonymize!
       RedeemifyCode.anonymize! current_user
       VendorCode.anonymize_all! current_user
-
       session.delete(:user_id)
-      redirect_to root_url, :flash => { :notice => "Your account has been deleted." }
+      redirect_to root_url, notice: "Your account has been deleted"
     end
   end
 
-
   def destroy
-    session[:user_id] = nil
-    session[:vendor_id] = nil
-    session[:provider_id] = nil
+    session.delete(:user_id)
+    session.delete(:vendor_id)
+    session.delete(:provider_id)
     gon.clear
     redirect_to root_url, notice: "Signed out!"
   end
 
-
-
   def failure
-    redirect_to root_url, alert: "Authentication failed, please try again."
+    redirect_to root_url, alert: "Authentication failed, please try again"
   end
 
   def change_to_vendor
     session[:user_id] = nil
-    redirect_to '/vendors/home', notice: "Change to Vendor Account"
+    redirect_to '/vendors/home', notice: "Changed to vendor account"
   end
 
-  def hello_user
-  end
+  private
 
-  def welcome_user
-  end
-
-  def enter_user
-  end
-
-  def exit_user
+  def offeror
+    auth = request.env["omniauth.auth"]
+    Provider.find_by_provider_and_email(auth["provider"], auth["info"]["email"]) || 
+    Vendor.find_by_provider_and_email(auth["provider"], auth["info"]["email"])
   end
 
-  def login_user
+  def log_in(user)
+    key = "#{user.class.to_s.downcase}_id".to_sym
+    session[key] = user.id
   end
 
-  def logout_user
+  def find_or_create_user
+    auth = request.env["omniauth.auth"]
+    User.find_by_provider_and_email(auth["provider"], auth["uid"]) ||
+    User.create_with_omniauth(auth)
```

Comment:
This would seem like a typo:
the user is searched by_provider_and_email with auth["uid"]) for the email argument. 
The prior version featured
User.find_by_provider_and_**uid**(auth["provider"], auth["uid"]) .